tests/data_context/cloud_data_context/test_checkpoint_crud.py

```diff
@@ -0,0 +1,204 @@
+            },
+            200,
+        )
+
+    return _mocked_get_response
+
+
+@pytest.mark.cloud
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "data_context_fixture_name,data_context_type",
+    [
+        # In order to leverage existing fixtures in parametrization, we provide
+        # their string names and dynamically retrieve them using pytest's built-in
+        # `request` fixture.
+        # Source: https://stackoverflow.com/a/64348247
+        pytest.param(
+            "empty_base_data_context_in_cloud_mode",
+            BaseDataContext,
+            id="BaseDataContext",
+        ),
+        pytest.param("empty_data_context_in_cloud_mode", DataContext, id="DataContext"),
+        pytest.param(
+            "empty_cloud_data_context", CloudDataContext, id="CloudDataContext"
+        ),
+    ],
+)
+def test_cloud_backed_data_context_add_checkpoint(
+    data_context_fixture_name: str,
+    data_context_type: Type[AbstractDataContext],
+    checkpoint_id: str,
+    validation_ids: Tuple[str, str],
+    checkpoint_config: dict,
+    mocked_post_response: Callable[[], MockResponse],
+    mocked_get_response: Callable[[], MockResponse],
+    request_headers: Dict[str, str],
+    ge_cloud_base_url: str,
+    ge_cloud_organization_id: str,
+    request,
+) -> None:
+    """
+    All Cloud-backed contexts (DataContext, BaseDataContext, and CloudDataContext) should save to a Cloud-backed CheckpointStore when calling `add_checkpoint`.
+    When saving, it should use the id from the response to create the checkpoint.
+    """
+    context = request.getfixturevalue(data_context_fixture_name)
+
+    # Make sure the fixture has the right configuration
+    assert isinstance(context, data_context_type)
+    assert context.ge_cloud_mode
+
+    validation_id_1, validation_id_2 = validation_ids
+
+    with mock.patch(
+        "requests.post", autospec=True, side_effect=mocked_post_response
+    ) as mock_post, mock.patch(
+        "requests.get", autospec=True, side_effect=mocked_get_response
+    ) as mock_get:
+        checkpoint = context.add_checkpoint(**checkpoint_config)
+
+        # Round trip through schema to mimic updates made during store serialization process
+        expected_checkpoint_config = checkpointConfigSchema.dump(
+            CheckpointConfig(**checkpoint_config)
+        )
+
+        mock_post.assert_called_with(
+            f"{ge_cloud_base_url}/organizations/{ge_cloud_organization_id}/contracts",
+            json={
+                "data": {
+                    "type": "contract",
+                    "attributes": {
+                        "checkpoint_config": expected_checkpoint_config,
+                        "organization_id": ge_cloud_organization_id,
+                    },
+                },
+            },
+            headers=request_headers,
+        )
+
+        mock_get.assert_called_with(
+            f"{ge_cloud_base_url}/organizations/{ge_cloud_organization_id}/contracts/",
```

Comment:
I think this should have the checkpoint id in the second call rather than just to the `/contracts/` endpoint so we get the specific checkpoint.